presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -221,7 +222,10 @@ LongEnumMap getLongEnumMap()
         VarcharEnumMap getVarcharEnumMap()
         {
             checkArgument(!isLongEnum, "Invalid enum map format");
-            return new VarcharEnumMap(map);
+            // Varchar enum values are base32-encoded so that they are case-insensitive, which is expected of TypeSigntures
+            Base32 base32 = new Base32();
+            return new VarcharEnumMap(map.entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> new String(base32.decode(e.getValue().toUpperCase(ENGLISH))))));
```

Comment:
why not to keep it in lower case?

Reply:
let me check but I believe the base32 decoder expects uppercase letters in its encoding